### ISSUE TITLE
Adding an incident using the API - A user with that username already exists.

### ISSUE THREAD

**Issue by lnxg33k** [certsocietegenerale/FIR#68]
I am trying to add an incident using the API and I get, 400 status code with the following message 
**{"opened_by":{"username":["A user with that username already exists."]}}**

Here is the json object I am trying to post
```
{'status': 'O', 'description': 'bam bam beet', 'file_set': [], 'is_major': False, 'detection': 1, 'is_incident': False, 'date': '2016-02-09T14:08:00', 'is_starred': False, 'subject': 'lolaaaa', 'confidentiality': 1, 'severity': 1, 'opened_by': {'username': 'lnxg33k'}}
```

I am using Python requests to interact with the API:

`x = request("POST", "http://127.0.0.1:8000/api/incidents", headers={'Authorization': 'Token 508ca6d83242b27c584ef03bcebb62e912b88681'}, json=data)`

**Comment by eth-0** [comment]
I'm seeing the same issue.

Here's my code:

`import requests
import json

url = 'http://localhost:8000/api/incidents'
payload = {'subject':'API','description':'description','detection':'17','file_set':[],'opened_by':{'username':'user',},'severity':'4'}
headers = {'content-type': 'application/json', 'Authorization': 'Token XXXXXXXXXXXXXXXXXXXXXXXXXXXX'}

response = requests.post(url, data=json.dumps(payload), headers=headers)
print response.content
`

**Comment by lnxg33k** [comment]
Hey @eth-0,  I solved it with a dirty _workaround_ till the devs solve the issue and document the API plugin.

Just replace **IncidentSerializer** class with the following one:

```python
class IncidentSerializer(serializers.ModelSerializer):
    detection = serializers.PrimaryKeyRelatedField(queryset=Label.objects.filter(group__name='detection'))
    file_set = AttachedFileSerializer(many=True)

    class Meta:
        model = Incident
        exclude = ['main_business_lines', 'opened_by', 'plan', 'actor', 'file_set']
        read_only_fields = ('id', 'main_business_lines', 'artifacts', 'file_set')
```

**Comment by eth-0** [comment]
Hey thanks @lnxg33k. I was trying to fix it myself as well, but I got lost too lost in the code. I'll try it out!

**Comment by tomchop** [comment]
Thanks @lnxg33k, the workaround makes sense. With some extra work, we came up with something that worked as expected (no longer need to specify an `opened_by` field since this is supposed to be bound by the token)

```json
POST /api/incidents HTTP/1.1
Accept: application/json
Accept-Encoding: gzip, deflate
Authorization: Token 5301cd1ba9180e3fd33b8df688f3c948ecfadb77
Connection: keep-alive
Content-Length: 256
Content-Type: application/json
Host: localhost:8000
User-Agent: HTTPie/0.9.3

{
    "category": "3",
    "concerned_business_lines": [
        1
    ],
    "confidentiality": "1",
    "date": "2016-02-10T20:37:17",
    "description": "Description",
    "detection": "2",
    "is_incident": "true",
    "is_major": "true",
    "severity": "1",
    "status": "C",
    "subject": "Test incident"
}

HTTP/1.0 201 Created
Allow: GET, POST, HEAD, OPTIONS
Content-Language: en
Content-Type: application/json
Date: Wed, 10 Feb 2016 20:27:39 GMT
Server: WSGIServer/0.1 Python/2.7.11
Vary: Accept, Accept-Language, Cookie

{
    "actor": null,
    "category": 3,
    "concerned_business_lines": [
        1
    ],
    "confidentiality": 1,
    "date": "2016-02-10T20:37:17",
    "description": "Description",
    "detection": 2,
    "file_set": [],
    "id": 9,
    "is_incident": true,
    "is_major": true,
    "is_starred": false,
    "opened_by": 1,
    "plan": null,
    "severity": 1,
    "status": "C",
    "subject": "Test incident"
}
```

As you can see, the returned object is "flat" (ie. not loading embedded fields anymore). This might not be an issue if that kind of information is previously loaded or queried as need be. Thoughts?

**Comment by lnxg33k** [comment]
@tomchop Yeah it is cool and as you said it will make sense to exclude the opened_by as it should be auto bounded with the authenticated user, hope will see it in the master branch soon.

**Comment by eth-0** [comment]
When can we expect to see this in the mast branch?

**Comment by tomchop** [comment]
Yes, I'll try make some time to merge this by next week.

**Comment by kevthehermit** [comment]
Why is this closed? Its still an issue.

**Comment by tomchop** [comment]
@kevthehermit: You're right, it should still be open until we merge this into master. Sorry about that. I'll be closing your other issue and leave this one open if you don't mind.

**Comment by sroberts** [comment]
If the PR references this issue it will automatically be closed re: https://github.com/blog/1506-closing-issues-via-pull-requests.

**Comment by tomchop** [comment]
I tried referencing the issue but failed. This should be fixed in de0b11134cfdfa32c86ca24ae6e8eabe28383806.

Here's a sample request / response of how incidents are now created.

```json
POST /api/incidents HTTP/1.1
Accept: application/json
Accept-Encoding: gzip, deflate
Authorization: Token abcdabcdabcdabcdabcdabcdabcdabcdabcdabcd
Connection: keep-alive
Content-Length: 256
Content-Type: application/json
Host: localhost:8000
User-Agent: HTTPie/0.9.3

{
    "category": "3",
    "concerned_business_lines": [
        1
    ],
    "confidentiality": "1",
    "date": "2016-02-10T20:37:17",
    "description": "Description",
    "detection": "2",
    "is_incident": "true",
    "is_major": "true",
    "severity": "1",
    "status": "C",
    "subject": "Test incident"
}

HTTP/1.0 201 Created
Allow: GET, POST, HEAD, OPTIONS
Content-Language: en
Content-Type: application/json
Date: Thu, 10 Mar 2016 17:25:28 GMT
Server: WSGIServer/0.1 Python/2.7.11
Vary: Accept, Accept-Language, Cookie

{
    "actor": null,
    "category": 3,
    "concerned_business_lines": [
        1
    ],
    "confidentiality": 1,
    "date": "2016-02-10T20:37:17",
    "description": "Description",
    "detection": 2,
    "file_set": [],
    "id": 12,
    "is_incident": true,
    "is_major": true,
    "is_starred": false,
    "opened_by": 1,
    "plan": null,
    "severity": 1,
    "status": "C",
    "subject": "Test incident"
}
```
